plugin-server/src/worker/ingestion/event-pipeline/7-runAsyncHandlersStep.ts

```diff
@@ -20,15 +20,13 @@ export async function runAsyncHandlersStep(
 
 async function processOnEvent(runner: EventPipelineRunner, event: PostIngestionEvent) {
     const processedPluginEvent = convertToProcessedPluginEvent(event)
-    const isSnapshot = event.event === '$snapshot'
```

Comment:
doesn't the runAsyncHandlersStep come from a separate kafka topic potentially, so we'd need to make sure we handle its similarly to plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

Reply:
It's not and hasn't been for some time. With the initial breaking up of plugin-server into ingestion and async we ended up not running onSnapshot at all, as async only reads from `clickhouse_events_json` and not `clickhouse_session_recording_events`

Reply:
should we nuke that functionality given no-one complained? Can we look at who's using it?

Reply:
Yeah I'm gonna just remove the implementation and tests here, and remove the references on https://github.com/PostHog/posthog.com/blob/master/contents/docs/apps/build/reference.md

Reply:
https://github.com/PostHog/posthog.com/pull/5159